internal/command/volumes/create.go

```diff
@@ -76,17 +77,25 @@ func runCreate(ctx context.Context) error {
 		appName    = appconfig.NameFromContext(ctx)
 	)
 
-	app, err := client.GetAppBasic(ctx, appName)
-	if err != nil {
-		return err
-	}
+	// pre-fetch platform regions from API in background
+	prompt.PlatformRegions(ctx)
+
+	// fetch AppBasic in the background while we prompt for confirmation
+	appFuture := future.Spawn(func() (*api.AppBasic, error) {
+		return client.GetAppBasic(ctx, appName)
+	})
```

Comment:
hey, the improvements are nice but I have mixed feelings about using Future/Promise pattern in a Go codebase. Somehow I tend to think it can be solved with Go primitives like a go routine and a channel.

Reply:
Using a goroutine and a channel was my first thought, but I ran into a few problems with that approach:

* A channel would only allow for a single receiver of the value. This would necessitate a different approach when pre-loading the list of regions in `platform.Region`, which could be called any number of times. A future neatly encapsulates the concept of a value that may or may not be ready yet, which can be passed around just like any other Go value.

* Error handling is clumsy with a single channel. We'd either need to wrap both success type and error type up in a single value (`interface{}`?) and then pull them out on the other end, or use two channels and `select` on the other end. Either way, it would be a lot of code that could be error-prone to write out each time. Future's API is difficult to use wrong and lets the logic, not the logistics, shine through.

* A panicking goroutine would dump a stack trace to the console, but it otherwise leave the waiting goroutines hanging without further special care. There's a few lines of code to deal with this case in `internal/future/future.go`, and my hope is that a small, neat abstraction can do the right thing for us in one place so we don't have to think about carefully handling this everywhere we'd like to repeat this pattern.

Goroutines and channels are useful, but they often aren't the whole part of the picture. For example, goroutines and channels pair very well with `WaitGroup` when writing code that fans out to multiple worker goroutines. You could implement the logic to keep count of how many goroutines are still running yourself, but doing so in every instance would be error-prone and this is a common enough pattern that the Go designers encapsulated this pattern into a general purpose abstraction.

My hope is that Future can serve as a useful concurrency primitive for us in a way similar to other Go stdlib primitives like `WaitGroup` or `Mutex` (in fact, the Future implementation here is essentially just a wrapper over RWMutex). There are cases where Future as a primitive is the best suited tool for the job, and I hope this PR does a good job of demonstrating a few of those cases.